tests/Doctrine/Tests/ORM/Functional/Ticket/GH7854Test.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional\Ticket;
+
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+/**
+ * @group GH7854
+ */
+class GH7854Test extends OrmFunctionalTestCase
+{
+    public function setUp()
+    {
+        parent::setUp();
+
+        $this->_schemaTool->createSchema(
+            [
+                $this->_em->getClassMetadata(GH7854Entity::class),
+                $this->_em->getClassMetadata(GH7854ValueObject::class),
+            ]
+        );
+    }
+
+    public function testTypedPropertyContainingEmbeddable() : void
+    {
+        $this->_em->persist(new GH7854Entity());
+        $this->_em->flush();
+        $this->_em->getRepository(GH7854Entity::class)->findAll();
+    }
+}
+
+/**
+ * @Entity()
+ */
+class GH7854Entity
+{
+    /**
+     * @Embedded(class = "GH7854ValueObject")
```

Comment:
This should use `::class` syntax instead

```suggestion
     * @Embedded(class=GH7854ValueObject::class)
```